test/integration/service_discovery_test.exs

```diff
@@ -0,0 +1,45 @@
+defmodule Pulsar.Integration.ServiceDiscoveryTest do
+  use ExUnit.Case
+  import TelemetryTest
+
+  require Logger
+
+  alias Pulsar.Test.Support.System
+  alias Pulsar.Test.Support.Utils
+  alias Pulsar.ServiceDiscovery
+
+  @moduletag :integration
+  @topic "persistent://public/default/service-discovery-test-topic"
+
+  setup do
+    broker = System.broker()
+
+    config = [
+      host: broker.service_url
+    ]
+
+    {:ok, app_pid} = Pulsar.start(config)
+
+    on_exit(fn ->
+      Process.exit(app_pid, :shutdown)
+      Utils.wait_for(fn -> not Process.alive?(app_pid) end)
+    end)
+  end
+
+  setup [:telemetry_listen]
+
+  describe "Topic Lookup" do
+    @tag telemetry_listen: [[:pulsar, :service_discovery, :lookup_topic, :stop]]
+    test "topic lookup retrieves broker PID" do
+      topic = @topic <> "-#{:erlang.unique_integer([:positive])}"
+
+      {:ok, broker_pid} = ServiceDiscovery.lookup_topic(topic)
+
+      assert Process.alive?(broker_pid)
+
+      # Verify 1 lookup with success result
+      stats = Utils.collect_lookup_stats()
+      assert %{success_count: 1, failure_count: 0, total_count: 1} = stats
```

Comment:
In this particular case, though - I think it would be best for the test to assert on the actual result from `ServiceDiscovery.lookup_topic/1` (or both!). For example, I don't understand how this test succeeds 🙃 The topic used in the `lookup_topic` does not exist. Thus, I would have expected `ServiceDiscovery.lookup_topic/1` to return `{:error, reason}`. But maybe you inherited a bogus implementation of the topic lookup flow. I realized I never wrote any specific tests for it.

Reply:
I actually had a test for the error case but it kept not giving errors and I assumed there was a topic creation on lookup or something. Didn't have much time to look into it sorry! 

For the metrics I saw that in the go client they do have a lookup request metric event so I went for it (and also the java one iirc). But you are right I should also check on the result itself (I was doing it then removed it).

Reply:
> I actually had a test for the error case but it kept not giving errors and I assumed there was a topic creation on lookup or something. Didn't have much time to look into it sorry!

I will try to dig out more details. I'm curious to know what's going on there 😄

Reply:
> Then it should good to go?

I would say so! 😄 Oh, and I suggest updating the `Consumer` module to use the new `ServiceDiscovery` module too ([see this other comment](https://github.com/efcasado/pulsar-elixir/pull/24#issuecomment-3418452790)).

Once again, thanks a lot for your contributions! 🙌